app/src/js/components/DataUpload/uploadStep.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter, useParams } from 'react-router-dom';
+import { connect, useDispatch } from 'react-redux';
+import Breadcrumbs from '../Breadcrumbs/Breadcrumbs';
+import queryString from 'query-string';
+import UploadOverview from '../DataUpload/overview';
+import { getUploadStep, promoteStep } from '../../actions';
+import { Alert } from 'react-bootstrap';
+
+const formatStepName = (stepName) => {
+  return stepName.replace(/_/g, ' ').replace(/\b\w/g, (l) => l.toUpperCase());
+};
+
+const breadcrumbConfig = [
+  {
+    label: 'Dashboard Home',
+    href: '/'
+  },
+  {
+    label: 'Upload',
+    active: true
+  }
+];
+
+const UploadStep = ({ privileges, location, history}) => {
+  const dispatch = useDispatch();
+  const { requestId } = useParams();
+  const { uploadStepId } = queryString.parse(location.search);
+  const [stepData, setStepData] = useState({});
+  const [allowUpload, setAllowUpload] = useState(false);
+  const [alertVariant, setAlertVariant] = useState('success');
+  const [alertMessage, setAlertMessage] = useState('');
+  const [dismissCountDown, setDismissCountDown] = useState(0);
+
+  useEffect(() => {
+    if (uploadStepId) {
+      dispatch(getUploadStep(uploadStepId))
+        .then(({ data }) => {
+          if (data.error) {
+            setAlertVariant('danger');
+            setAlertMessage('Operation failed due to unexpected database error.');
+            setDismissCountDown(10);
+          } else {
+            data.step_name = formatStepName(data.step_name);
+            setStepData(data);
+            setAllowUpload(true);
+          }
+        });
+    }
+  }, [dispatch, uploadStepId]);
+
+  useEffect(() => {
+    if (dismissCountDown > 0) {
+      window.scrollTo(0, 0);
+      const intervalId = setInterval(() => {
+        setDismissCountDown((prevCount) => prevCount - 1);
+      }, 1000);
+
+      return () => clearInterval(intervalId);
+    }
+  }, [dismissCountDown]);
+
+  const navigateBack = () => {
+    history.push('/requests');
+  }
+
+  const handleSubmit = async () => {
+    await dispatch(promoteStep({ id: requestId }))
+      .then(() => {
+        history.push('/requests');
```

Comment:
I noticed that this refresh seems to be happening "too fast" so that the redirect is happening, but the home page is still showing the old step. Nothing too major, so I'll just make a note to look more into it later.